### ISSUE TITLE
The length of labels must equal to the number of rows in the input data

### ISSUE THREAD

**Issue by spsanderson** [business-science/modeltime#123]
Here is my time-series data: 
[data_tbl.xlsx](https://github.com/business-science/modeltime/files/6939186/data_tbl.xlsx)

Here is my juiced data (recipe_num_only recipe):
[juiced_recipe.xlsx](https://github.com/business-science/modeltime/files/6939188/juiced_recipe.xlsx)

Here is my splits object:
```r
splits <- initial_time_split(
  data_final_tbl
  , prop = 0.8
  , cumulative = TRUE
)
```

Here are my recipes (the one in question is recipe_num_only)

```r
# Features ----------------------------------------------------------------

recipe_base <- recipe(value ~ ., data = training(splits))

recipe_date <- recipe_base %>%
  step_timeseries_signature(date_col) %>%
  step_rm(matches("(iso$)|(xts$)|(hour)|(min)|(sec)|(am.pm)")) %>%
  step_normalize(contains("index.num"), contains("date_col_year"))

recipe_fourier <- recipe_date %>%
  step_dummy(all_nominal_predictors(), one_hot = TRUE) %>%
  step_fourier(date_col, period = 365/12, K = 1) %>%
  step_YeoJohnson(value, limits = c(0,1))

recipe_fourier_final <- recipe_fourier %>%
  step_nzv(all_predictors())

recipe_pca <- recipe_base %>%
  step_timeseries_signature(date_col) %>%
  step_rm(matches("(iso$)|(xts$)|(hour)|(min)|(sec)|(am.pm)")) %>%
  step_dummy(all_nominal_predictors(), one_hot = TRUE) %>%
  step_normalize(value) %>%
  step_fourier(date_col, period = 365/52, K = 1) %>%
  step_normalize(all_numeric_predictors()) %>%
  step_nzv(all_predictors()) %>%
  step_pca(
    all_numeric_predictors(), -date_col_index.num
    , threshold = .95
  )

recipe_num_only <- recipe_pca %>%
  step_rm(-all_numeric_predictors())
```

Here is my XGBoost Model Spec

```r
# XGBoost -----------------------------------------------------------------

model_spec_boost <- boost_tree(
  mode  = "regression"
  # , mtry  = 25
  # , trees = 25
  # , min_n = 10
  # , tree_depth = 2
  # , learn_rate = 0.3
  # , loss_reduction = 0.01
) %>%
  set_engine("xgboost")

# * * Testing ----
set.seed(123)
workflow() %>%
  add_model(model_spec_boost) %>%
  add_recipe(recipe_num_only) %>%
  fit(training(splits))
# * * End Test ----
```

The error I get is the following:
```r
> workflow() %>%
+   add_model(model_spec_boost) %>%
+   add_recipe(recipe_num_only) %>%
+   fit(training(splits))
Error in setinfo.xgb.DMatrix(dmat, names(p), p[[1]]) : 
  The length of labels must equal to the number of rows in the input data
Timing stopped at: 0 0 0
```

Everything works until I get to the `fit(training(splits))`

```r
> workflow() %>%
+   add_model(model_spec_boost) %>%
+   add_recipe(recipe_num_only)
== Workflow ==========================================================================================
Preprocessor: Recipe
Model: boost_tree()

-- Preprocessor --------------------------------------------------------------------------------------
9 Recipe Steps

* step_timeseries_signature()
* step_rm()
* step_dummy()
* step_normalize()
* step_fourier()
* step_normalize()
* step_nzv()
* step_pca()
* step_rm()

-- Model ---------------------------------------------------------------------------------------------
Boosted Tree Model Specification (regression)

Computational engine: xgboost 
```

At a bit of a loss here

**Comment by mdancho84** [comment]
I'd try to investigate your recipe transformation. Xgboost is picky with data types. 

Try applying your recipe to the training data and report back with the data transformation and data types.

**Comment by spsanderson** [comment]
The juiced_data is the recipe applied to the training data and it is all numeric.

**Comment by mdancho84** [comment]
I found this stack overflow https://stackoverflow.com/questions/35852923/xgb-dmatrix-error-the-length-of-labels-must-equal-to-the-number-of-rows-in-the

I suspect some of your numeric data is constant. Maybe add `step_zv()`?

**Comment by spsanderson** [comment]
nope, my recipe had a `step_nzv` so I changed to `step_zv` as indicated and same exact issue, any idea on other ways to debug?

**Comment by mdancho84** [comment]
You removed your target. Change your `recipe_num_only` to this:

```
recipe_num_only <- recipe_pca %>%
  step_rm(-value, -all_numeric_predictors())
```

**Comment by spsanderson** [comment]
👏
